src/langpacks.py

```diff
@@ -0,0 +1,332 @@
+                raise
+            except PackageError as e:
+                logger.error(f"failed to install {p}: {e}")
+                raise
+
+        # Clone the langpack-o-matic repo
+        try:
+            run(
+                [
+                    "sudo",
+                    "-u",
+                    "ubuntu",
+                    "git",
+                    "clone",
+                    "-b",
+                    "master",
+                    "https://git.launchpad.net/langpack-o-matic",
+                    REPO_LOCATION,
+                ],
+                check=True,
+                stdout=PIPE,
+                stderr=STDOUT,
+                text=True,
+            )
+            logger.debug("Langpack-o-matic vcs cloned.")
+        except CalledProcessError as e:
+            logger.debug(f"Git clone of the code failed: {e.stdout}")
+            raise
+
+    def update_checkout(self):
+        """Update the langpack-o-matic checkout."""
+        # Pull Vcs updates
+        try:
+            run(
+                [
+                    "sudo",
+                    "-u",
+                    "ubuntu",
+                    "git",
+                    "-C",
+                    REPO_LOCATION,
+                    "pull",
+                ],
+                check=True,
+                stdout=PIPE,
+                stderr=STDOUT,
+                text=True,
+            )
+            logger.debug("Langpack-o-matic checkout updated.")
+        except CalledProcessError as e:
+            logger.debug(f"Git pull of the langpack-o-matic repository failed: {e.stdout}")
+            raise
+
+        # Call make target
+        try:
+            run(
+                [
+                    "sudo",
+                    "-u",
+                    "ubuntu",
+                    "make",
+                    "-C",
+                    REPO_LOCATION / "bin",
+                ],
+                check=True,
+                stdout=PIPE,
+                stderr=STDOUT,
+                text=True,
+            )
+            logger.debug("Langpack-o-matic bin/msgequal build.")
+        except CalledProcessError as e:
+            logger.debug(f"Build of bin/msgequal failed: {e.stdout}")
+            raise
+
+    def _clean_builddir(self, releasedir):
+        """Clean build cache."""
+        if not os.path.exists(releasedir):
+            return
+        for builddir in (
+            releasedir / "sources-base",
+            releasedir / "sources-update",
+        ):
+            if os.path.exists(builddir):
+                try:
+                    shutil.rmtree(builddir)
+                    logger.debug(f"Removed the existing cache directory: {builddir}")
+                except OSError as e:
+                    logger.error(f"Failed to remove cache directory {builddir}: {e}")
+
+    def build_langpacks(self, base, release):
+        """Build the langpacks."""
+        lp = Launchpad.login_anonymously("langpacks", "production")
```

Comment:
You may want to dependency-inject an LP client you own here to make testing this class easier (similar to https://github.com/canonical/ubuntu-manpages-operator/blob/main/src/launchpad.py)